ai2thor/tests/test_unity.py

```diff
@@ -80,6 +80,25 @@ def test_small_aspect():
     event = controller.step(dict(action='Initialize', gridSize=0.25))
     assert event.frame.shape == (64, 128, 3)
 
+@pytest.mark.parametrize("controller", [fifo_controller])
+def test_fast_emit(controller):
+    event = controller.step(dict(action='RotateRight'))
+    event_fast_emit = controller.step(dict(action='TestFastEmit', rvalue='foo'))
+    # assert that when actionFastEmit is off that the objects are different
+    assert id(event.metadata['objects']) !=  id(event_fast_emit.metadata['objects'])
```

Comment:
why are they different if the `TestFastEmit` action does not change the camera environment or anything? there is another metadata generation vs a metadata patch but shouldn't it be the same metadata generated?

Reply:
I refactored the tests into two separate tests - note the comment above the comparison.  They are different in the case when fastActionEmit is disabled.